### PR TITLE
Add an ansible-runner test case

### PR DESCRIPTION
**Pull request by gandelman-a**
This adds a test case for the ansible-runner script which:

* Creates an upstream git repo containing a playbook and a requirements.txt
* Clones the upstream, runs the ansible-runner
* Asserts playbook is run and that requirements have been installed
* Updates the upstream playbook, runs ansible-runner
* Asserts updates have been pulled down and the playbook has been run
* Adds a new requirement, runs the ansible-runner
* Asserts the new requirement has been installed into expected venv

This required removing a bunch of hard-coded paths from the ansible-runner
script.

Also, organized the tests/ directory a bit and moved non-script things into
a files/ subdir.

Resolves-Issue: BonnyCI/projman#163

### CODE REVIEW

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/215/1489179290.32/bonnyci-run-check-multinode : FAILURE in 57s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/215/1489179358.82/bonnyci-run-check-multinode : FAILURE in 56s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/215/1489181900.4/bonnyci-run-check-multinode : FAILURE in 20s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/215/1489184201.44/bonnyci-run-check-multinode : FAILURE in 20s

**Comment by anne-bonny** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

File: roles/ansible-runner/files/usr/local/bin/ansible-runner [hunk 1]
```diff
@@ -3,7 +3,18 @@
 set -eu
 
 env=$1
-. /etc/default/$env
+envs_root=${ANSIBLE_RUNNER_ENV_ROOT:-/opt/source}
+ansible_venv=${ANSIBLE_RUNNER_VENV:-/opt/ansible}
+log_dir=${ANSIBLE_RUNNER_LOG_DIR:-/var/www/html/cron-logs/$env/}
+
+if [ -f /etc/default/env ]; then
```
Comment (ConsoleCatzirl) [inline roles/ansible-runner/files/usr/local/bin/ansible-runner]: Typo: `/etc/default/env` should be `/etc/default/$env`

**Review by ConsoleCatzirl — CHANGES_REQUESTED** [review]
There's a typo that needs to be fixed. But also, why does this need the overhead of a git repo? What do we gain by committing test files into a throwaway repo?

**Comment by gandelman-a** [comment]
So one of the core functions of the script is to pull down new revisions into a local checkout and apply those new ansible and requirement changes.  Our test should validate that this actually works. IIRC there were some instances where we were running ansible on outdated repos. 

Also, we require a git checkout for the script to even run.

File: roles/ansible-runner/files/usr/local/bin/ansible-runner (hunk 1 above)
Comment (gandelman-a) [reply roles/ansible-runner/files/usr/local/bin/ansible-runner]: Doh

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/215/1489427120.68/bonnyci-run-check-multinode : FAILURE in 1m 03s

**Comment by gandelman-a** [comment]
recheck

**Review by omgjlk — APPROVED** [review]
Looks good. Yay for more coverage.

**Review by ConsoleCatzirl — APPROVED** [review]

**Comment by ConsoleCatzirl** [comment]
gate-recheck

**Pull request merged**